oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexConfigTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.index.lucene;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.jackrabbit.oak.spi.state.NodeBuilder;
+import org.apache.jackrabbit.oak.spi.state.NodeState;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.Properties;
+import java.util.Set;
+
+import static com.google.common.collect.Sets.newHashSet;
+import static org.apache.jackrabbit.oak.InitialContentHelper.INITIAL_CONTENT;
+import static org.apache.jackrabbit.oak.plugins.index.lucene.writer.IndexWriterUtils.getIndexWriterConfig;
+import static org.junit.Assert.assertEquals;
+
+public class LuceneIndexConfigTest {
+
+    private NodeState root;// = INITIAL_CONTENT;
+
+    private Set<File> dirs = newHashSet();
+
+    private Properties properties;
+
+    @Before
+    public void before() {
+        properties = (Properties) System.getProperties().clone();
+        root = INITIAL_CONTENT;
+        System.setProperty("oak.index.lucene.maxBufferedDeleteTerms", "1000");
+        System.setProperty("oak.index.lucene.ramPerThreadHardLimitMB", "100");
+    }
+
+    @After
+    public void after() {
+        System.setProperties(properties);
+        for (File d : dirs) {
+            FileUtils.deleteQuietly(d);
+        }
+    }
```

Comment:
Could we use this Junit4 extension https://stefanbirkner.github.io/system-rules/#ProvideSystemProperty to safely deal with system properties?

Reply:
Done